recover-disk draft (#3496)
yugabyte/yugabyte-db: docs/content/latest/troubleshoot/nodes/recover-disk.md

---

> title: Disk Failure

Reviewer: ```suggestion
title: Disk failure
```

> linkTitle: Disk Failure

Reviewer: ```suggestion
linkTitle: Disk failure
```

description: Recover failing disk
aliases:
  - /troubleshoot/nodes/disk-failure/
menu:
  latest:
    parent: troubleshoot-nodes
    weight: 849
isTocNested: true
showAsideToc: true
---

> YugabyteDB can be configured to use multiple disks with the [`--fs_data_dirs`](../../reference/configuration/yb-tserver.md) gflag.

Reviewer: ```suggestion
YugabyteDB can be configured to use multiple disks using the [`--fs_data_dirs`](../../reference/configuration/yb-tserver.md) configuration option.
```

> YugabyteDB can be configured to use multiple disks with the [`--fs_data_dirs`](../../reference/configuration/yb-tserver.md) gflag.

Reviewer: ```suggestion
YugabyteDB can be configured to use multiple storage disks by setting the [`--fs_data_dirs`](../../reference/configuration/yb-tserver.md) configuration option.
```

This introduces the possibility of disk failure and recovery issues.

## Cluster replication recovery

> The tserver automatically detects the disk failure and tries to spread the data on that disk to other healthy nodes in the cluster.

Reviewer: ```suggestion
The `yb-tserver` service automatically detects disk failures and attempts to spread the data from the failed disk to other healthy nodes in the cluster.
```

> In a single-zone setup with replication factor 3: if you started with 4 nodes or more, 

Reviewer: ```suggestion
In a single-zone setup with a replication factor (RF) of `3`: if you started with four nodes or more, 
```

then there would be at least three nodes left after one failed. 

> In this case, re-replication is automatically started if a tserver/disk is down for 10 minutes.

Reviewer: ```suggestion
In this case, rereplication is automatically started if a YB-TServer or disk is down for 10 minutes.
```

> In a multi-zone setup with replication-factor 3: YugabyteDB will try to keep one copy of data per zone. 

Reviewer: ```suggestion
In a multi-zone setup with a replication factor (RF) of `3`: YugabyteDB will try to keep one copy of data per zone. 
```

> In this case, for automatic re-replication of data, a zone needs to have at least two tservers so that if one fails, 

Reviewer: ```suggestion
In this case, for automatic rereplication of data, a zone needs to have at least two YB-TServers so that if one fails, 
```

> its data can be re-replicated to the other. Thus, this would mean at least a 6 node cluster.

Reviewer: ```suggestion
its data can be rereplicated to the other. Thus, this would mean at least a six-node cluster.
```

## Failed disk replacement

> The procedure to replace a failed disk is in order:

Reviewer: ```suggestion
The steps to replace a failed disk are:
```

> 1. Stop the tserver

Reviewer: ```suggestion
1. Stop the YB-TServer node.
```

> 2. Replace the disk(s) that have failed

Reviewer: ```suggestion
2. Replace the disks that have failed.
```

> 3. Restart the tserver

Reviewer: ```suggestion
3. Restart the `yb-tserver` service.
```

> On restart, the tserver will see the new empty disk and start replicating tablets from other nodes.

Reviewer: ```suggestion
On restart, the YB-TServer will see the new empty disk and start replicating tablets from other nodes.
```
